src/base/ApplicationECSService.ts

```diff
@@ -437,4 +465,96 @@ export class ApplicationECSService extends Resource {
       tags: { ...this.config.tags, type: name },
     });
   }
+
+  private createEfsMount(efsFs: EFSProps) {
+    const ingress: vpc.SecurityGroupIngress[] = [
+      {
+        fromPort: 2049,
+        protocol: 'TCP',
+        toPort: 2049,
+        securityGroups: [this.ecsSecurityGroup.id],
+        description: 'required',
+        cidrBlocks: [],
+        ipv6CidrBlocks: [],
+        prefixListIds: [],
+      },
+    ];
+
+    const egress: vpc.SecurityGroupEgress[] = [
+      {
+        fromPort: 0,
+        protocol: '-1',
+        toPort: 0,
+        cidrBlocks: ['0.0.0.0/0'],
+        description: 'required',
+        ipv6CidrBlocks: [],
+        prefixListIds: [],
+        securityGroups: [],
+      },
+    ];
+
+    const mountSecurityGroup = new vpc.SecurityGroup(this, 'efs_mount_sg', {
+      namePrefix: `${this.config.prefix}-ECSSMountPoint`,
+      description: 'ECS EFS Mount (Managed by Terraform)',
+      vpcId: this.config.vpcId,
+      ingress,
+      egress,
+      tags: this.config.tags,
+      lifecycle: {
+        createBeforeDestroy: true,
+      },
+    });
+
+    // https://developer.hashicorp.com/terraform/cdktf/concepts/iterators
+    const iterator = TerraformIterator.fromList(this.config.privateSubnetIds);
+
+    new efs.EfsMountTarget(this, 'efs_mount_target', {
+      forEach: iterator,
+      fileSystemId: efsFs.id,
+      subnetId: iterator.value,
+      securityGroups: [mountSecurityGroup.id],
+    });
+  }
+
+  private efsFilePolicy(
+    efsFs: EFSProps,
+    roleArn: string,
+    creationToken: string
+  ) {
+    const FsPolicy = {
+      Version: '2012-10-17',
+      Id: creationToken,
+      Statement: [
+        {
+          Sid: creationToken,
+          Effect: 'Allow',
+          Principal: {
+            AWS: roleArn,
+          },
+          Resource: efsFs.arn,
+          Action: [
+            'elasticfilesystem:ClientMount',
+            'elasticfilesystem:ClientWrite',
+          ],
+          Condition: {
+            Bool: {
+              'elasticfilesystem:AccessedViaMountTarget': 'true',
+            },
+          },
+        },
+      ],
+    };
+
+    const wait2Minutes = new Sleep(this, 'wait2Minutes', {
```

Comment:
nit pick: Should we rename this to `waitTwoMinutes`? Unless there's a good reason and I don't know about it 😅

Reply:
Updated pr with this suggestion